Contributing-errors.md

```diff
@@ -0,0 +1,163 @@
+# Error condition best practices
+
+## Advice about exposing errors in our APIs
+
+When updating code in this repository,
+consider the following best practices for exposing errors in our APIs.
+Depending on your contribution, you may alter code that might produce:
+
+- go error types and constants, suitable for use with [`errors.Is` and `errors.As`](https://pkg.go.dev/errors)
+- errors exposed over RPC (e.g. with [protocol buffers](https://grpc.io/docs/guides/error/))
+- translation of those proto errors into JSON and [HTTP status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) via REST endpoints
+- log messages [suitable for ops and SRE](https://www.honeycomb.io/blog/engineers-checklist-logging-best-practices), when contributing to service code
+- command line [status codes](https://tldp.org/LDP/abs/html/exitcodes.html)
+
+This document focuses on the first category, go `error` typed return values.
+
+### Concerns for Golang Library Developers
+
+- When a library does not provide explicitly exported and documented error types and constants,
+  application developers might depend on `strings.Contains()`
+  which bind their applications to the error message and prohibit improving error language
+  - Prefer using sentinel values for expected conditions
```

Comment:
an example of this might be nice, especially if we have an example in our code

Reply:
Looking at the code we don't do a good job of exporting or naming things. I think the only good example is the `sdk.ErrInvalid`, but even that is a little wrong (it should be `sdk.ErrReadInvalid` or `sdk.ErrTDFInvalid` following the rules below, since the SDK might also have invalid policy related requests).